sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,356 @@
+    return key !== Constants.XML_METADATA_MARKER;
+  });
+
+  if (contentElementNames && contentElementNames[0]) {
+    const contentRootElementName = contentElementNames[0];
+    delete entry.content[contentRootElementName][Constants.XML_METADATA_MARKER];
+    result = entry.content[contentRootElementName];
+
+    if (result) {
+      if (entry[Constants.XML_METADATA_MARKER]) {
+        result[Constants.ATOM_METADATA_MARKER] = entry[Constants.XML_METADATA_MARKER];
+      } else {
+        result[Constants.ATOM_METADATA_MARKER] = {};
+      }
+
+      result[Constants.ATOM_METADATA_MARKER]["ContentRootElement"] = contentRootElementName;
+
+      Object.keys(entry).forEach((property: string) => {
+        if (property !== "content" && property !== Constants.XML_METADATA_MARKER) {
+          result[Constants.ATOM_METADATA_MARKER][property] = entry[property];
+        }
+      });
+
+      return result;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * @ignore
+ * Utility to help parse given `feed` result
+ * @param feed
+ */
+function parseFeedResult(feed: any): object[] {
+  const result = [];
+  if (typeof feed === "object" && feed != null && feed.entry) {
+    if (Array.isArray(feed.entry)) {
+      feed.entry.forEach((entry: any) => {
+        const parsedEntryResult = parseEntryResult(entry);
+        if (parsedEntryResult) {
+          result.push(parsedEntryResult);
+        }
+      });
+    } else {
+      const parsedEntryResult = parseEntryResult(feed.entry);
+      if (parsedEntryResult) {
+        result.push(parsedEntryResult);
+      }
+    }
+  }
+  return result;
+}
+
+function isKnownResponseCode(
+  statusCode: number
+): statusCode is keyof typeof Constants.HttpResponseCodes {
+  return !!(Constants.HttpResponseCodes as { [statusCode: number]: string })[statusCode];
+}
+
+/**
+ * @ignore
+ * Extracts the applicable entity name(s) from the URL based on the known structure
+ * and instantiates the corresponding name properties to the deserialized response
+ *
+ * For instance, following is the URL structure for when creating a rule
+ * `<namespace-component>/<topic-name>/Subscriptions/<subscription-name>/Rules/<rule-name>`
+ *
+ * @param entry
+ * @param nameProperties
+ */
+function setName(entry: any, nameProperties: any): any {
+  if (entry[Constants.ATOM_METADATA_MARKER]) {
+    let rawUrl = entry[Constants.ATOM_METADATA_MARKER].id;
+
+    // The parsedUrl gets constructed differently for browser vs Node.
+    // It is specifically behaves different for some of the Atom based management API where
+    // the received URL in "id" element is of type "sb:// ... " and not a standard HTTP one
+    // Hence, normalizing the URL for parsing to work as expected in browser
+    if (rawUrl.startsWith("sb://")) {
+      rawUrl = "https://" + rawUrl.substring(5);
+    }
+
+    const parsedUrl = new URL(rawUrl);
+    const pathname: string = parsedUrl.pathname;
+    const parts = pathname.split("/");
```

Comment:
Should this ensure that the `parts` value is even?

I also assume that the rule name cannot contain a `/`

Reply:
Updated this and added comments inline.
Service allows `/` in queue and topic names, but not in namespace, subscription and rules